### PR TITLE
Improved README.md and add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by coderbot16**
People asking how to contribute has become a somewhat common question. Instead of having to repeat ourselves, it's better to just have a proper contribution guide!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+We use [MinecraftForge @ d28cd0352b6b0fe86e062f29e681c3b14572c6d5](https://github.com/MinecraftForge/MinecraftForge/tree/d28cd0352b6b0fe86e062f29e681c3b14572c6d5) as our reference commit. All features should be based on this version of Forge for consistency. For you convenience, ramidzkh has [remapped](https://github.com/PatchworkMC/YarnForge/tree/04d384add800bc395f4934507721f72eb733389f) Forge from MCP to Yarn: contributors should use this remapped version of Forge instead.
+
+This will be changed once we update to 1.15.
+
+
+## What is the general flow of contribution?
+
+1. Pick a feature to implement. There are a few ways to do this:
+    * Checking the `#wishlist` channel on the Discord server
```
Comment (TheGlitch76) [inline CONTRIBUTING.md:24]:
```suggestion
     * Checking the `#wishlist` channel on the [Discord server](https://discord.gg/YYZtNBG)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    * Scrolling through the [issue tracker](https://github.com/PatchworkMC/patchwork-api/issues)
+    * Picking something to implement yourself
+2. Locate mods to test the feature
```
Comment (TheGlitch76) [inline CONTRIBUTING.md:27]: something like "you can also make your own test mod (link to the repo) to isolate the feature, but please make sure that it is actually used in a mod"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    * If you're not sure what to do, ask on Discord in #api.
+    * In general, look at existing modules to see the naming conventions.
+4. Implement the feature and get it running.
+5. Clean up the code so that it adheres to Patchwork's standards.
+    * Remove [obvious lambda abuse](https://github.com/PatchworkMC/YarnForge/blob/04d384add800bc395f4934507721f72eb733389f/src/main/java/net/minecraftforge/fml/network/NetworkRegistry.java#L164-L180) and replace it with more readable code.
+    * Avoid [reckless usage](https://github.com/PatchworkMC/YarnForge/blob/04d384add800bc395f4934507721f72eb733389f/src/main/java/net/minecraftforge/fml/network/ICustomPacket.java) at runtime of Unsafe, Reflection, or ASM if possible.
+    * Clean up the JavaDoc to replace [MCP names](https://github.com/PatchworkMC/YarnForge/blob/04d384add800bc395f4934507721f72eb733389f/src/main/java/net/minecraftforge/event/entity/EntityEvent.java#L64-L76) with Yarn names. Also replace usage of `<br>` with proper paragraph tags.
+    * Make sure your code passes checkstyle by running `./gradlew checkstyleMain` (`gradlew.bat checkstyleMain` on Windows)
```
Comment (TheGlitch76) [inline CONTRIBUTING.md:38]:
```suggestion
    * Make sure your code passes checkstyle by running `gradlew checkStyleMain`
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+1. Pick a feature to implement. There are a few ways to do this:
+    * Checking the `#wishlist` channel on the Discord server
+    * Scrolling through the [issue tracker](https://github.com/PatchworkMC/patchwork-api/issues)
+    * Picking something to implement yourself
+2. Locate mods to test the feature
+    * It's important to make sure that you have a way to test what you're implementing. We all accidentally introduce bugs, and it's important to eliminate the more obvious ones through testing first.
+    * In addition, this also helps verify that the feature being implemented is actually used. Our time is limited, and it's good to make sure that we're spending it wisely.
+3. Determine the correct module to put the feature in, or make a new module.
+    * If you're not sure what to do, ask on Discord in #api.
```
Comment (TheGlitch76) [inline CONTRIBUTING.md:31]:
```suggestion
    * If you're not sure what to do, ask on [Discord](https://discord.gg/YYZtNBG) in #api.
```

File: README.md [hunk 5]
```diff
@@ -2,7 +2,21 @@
 
 A best-effort reimplementation of the Minecraft Forge API.
 
-## Contribution Notes
+## How do I use this?
 
-We use [MinecraftForge @ d28cd0352b6b0fe86e062f29e681c3b14572c6d5](https://github.com/MinecraftForge/MinecraftForge/tree/d28cd0352b6b0fe86e062f29e681c3b14572c6d5) as our reference commit. All features should be based on this version of Forge
-for consistency. This will be changed once we update to 1.15.
+Patchwork is not yet ready for general use. However, if you're ambitious and want to test, we provide prebuilt releases on [GitHub](https://github.com/PatchworkMC/patchwork-api/releases). Note that you will need to pass any Forge mods through [Patchwork Patcher](https://github.com/PatchworkMC/patchwork-patcher/releases) as well. While we do not expect most mods to work, bug reports would be appreciated.
+
```
Comment (TheGlitch76) [inline README.md:8]: add a link to our discord here and note that there's more information about compatibility there

File: README.md [hunk 6]
```diff
@@ -2,7 +2,21 @@
...
+If you know Java and would like to contribute, check out our [contribution guide](CONTRIBUTING.md)!
+
+
+## Technical details
+
+Patchwork API aims to re-implement the entirety of Minecraft Forge as a set of small and modular Fabric mods. This is in order to run Minecraft Forge mods patched using [Patchwork Patcher](https://github.com/PatchworkMC/patchwork-patcher) on Fabric.
+
+However, Patchwork aims not only to simply *run* Forge mods on Fabric but additionally aims to maintain a superior level of quality to Minecraft Forge. The Fabric toolchain is superior to Forge on a technical level, but lacks both the sheer quantity of hooks and features as well as existing mod support. Thus, Patchwork seeks to provide both the benefits of Fabric and the benefits of Forge.
```
Comment (TheGlitch76) [inline README.md:16]: I know none of us like Forge but let's not assert that fabric is always better. We both know there are cases where it isn't
```suggestion
However, Patchwork aims not only to simply *run* Forge mods on Fabric but additionally aims to maintain a superior level of code quality to Forge. We feel the Fabric toolchain is superior to Forge on a technical level, but lacks both the sheer quantity of hooks and features as well as existing mod support. Thus, Patchwork seeks to provide both the benefits of Fabric and the benefits of Forge.
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (coderbot16) [reply CONTRIBUTING.md:38]: I think it's alright to have simple clarifications here and there, especially for people who might not be super familiar with gradle overall.

**Pull request merged**